### PR TITLE
[Spec Decoding] Use correct shape for logit positions.

### PR DESCRIPTION
**Pull request by py4**
# Description

The number of speculative tokens can be more than total number of requests. We cannot use `num_padded_reqs` to use for logit positions. Because logit positions can go up to `num_reqs * (num_speculative_tokens + 1)`.
But this is not the case for bonus tokens. Each request, has only one bonus tokens. 

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_commons/runner/jax/tpu_jax_runner.py [hunk 1]
```diff
@@ -1244,20 +1258,25 @@ def _get_spec_decode_metadata(
...
         ])
+
+        assert bonus_logits_indices.shape[0] <= padded_num_reqs, (
+            f"bonus_logits_indices.shape[0]={bonus_logits_indices.shape[0]} "
+            f"padded_num_reqs={padded_num_reqs}")
+
         padded_bonus_logits_indices = np.concatenate([
             bonus_logits_indices,
-            np.zeros(padded_logits_length - bonus_logits_indices.shape[0],
+            np.zeros(padded_num_reqs - bonus_logits_indices.shape[0],
```
Comment (Lumosis) [inline tpu_commons/runner/jax/tpu_jax_runner.py:1274]: Have you tested this part with spec decoding enabled? I vaguely remembered we need to keep the padding length consistent. Otherwise, there might be some mismatch during sampling. (I could be wrong)

Comment (py4) [reply tpu_commons/runner/jax/tpu_jax_runner.py:1274]: @Lumosis Yeah I tested it on 1000 examples. The reason is that we have 1 bonus token per request, so the correct padding for `bonus_logits_indices` is up to `padded_num_reqs`
